### PR TITLE
Adding exponentiation by squaring to Monoid typeclass

### PR DESCRIPTION
**Pull request by cesartl**

### CODE REVIEW

**Comment by cesartl** [comment]
Could someone advise where I should add tests for this new method?

File: arrow-core-data/src/main/kotlin/arrow/typeclasses/Monoid.kt [hunk 1]
```diff
@@ -1,5 +1,7 @@
 package arrow.typeclasses
 
+import java.math.BigInteger
```
Comment (pakoito) [inline arrow-core-data/src/main/kotlin/arrow/typeclasses/Monoid.kt:3]: We cannot couple the library with Java, sadly.

**Comment by pakoito** [comment]
>/home/runner/work/arrow-core/arrow-core/arrow-core-data/src/main/kotlin/arrow/typeclasses/Monoid.kt:39:11: Unnecessary space(s)

The linter complained about an additional space in line 39. LGTM otherwise :D

**Review by pakoito — APPROVED** [review]

**Comment by aballano** [comment]
> Could someone advise where I should add tests for this new method?

You can add a generic test to our MonoidLaws, so it'll be automatically tested by all data types supporting Monoid

**Comment by rachelcarmena** [comment]
Hi fellow contributor :wave: 

Arrow repositories has been reunified again into [`arrow`](https://github.com/arrow-kt/arrow) so this repository is going to be archived.

Please, if you want to replay this pull request on `arrow` repository, follow these steps:

```
# Adapt your branch to be able to be merged in the new arrow repository
cd arrow-core
git-filter-repo --to-subdirectory-filter arrow-libs/core
cd ..

# Create a new branch in arrow repository and merge your content
git clone git@github.com:arrow-kt/arrow.git
cd arrow
git checkout -b <new-branch>
git remote add arrow-core ../arrow-core
git fetch arrow-core
git merge --no-edit --allow-unrelated-histories arrow-core/<your-branch>
# check and create the pull request again on arrow repository
```

Please, let us know if we can help you!

Thank you so much for your contributions!! :raised_hands:

**Pull request closed without merging**